jazz_asset-event-handler/index.js

```diff
@@ -0,0 +1,360 @@
+                    logger.info("Creating new asset records");
+                    processCreateAsset(eventPayload, configData, authToken)
+                        .then(result => { return resolve(result) })
+                        .catch(err => {
+                            logger.error("processCreateAsset Failed" + err);
+                            return reject(err)
+                        })
+                })
+
+        } else if (eventPayload.EVENT_NAME.S === configData.EVENTS.UPDATE_ASSET) {
+            checkIfAssetExists(eventPayload, configData, authToken)
+                .then(record => {
+                    processUpdateAsset(record, eventPayload, configData, authToken)
+                        .then(result => { return resolve(result) })
+                        .catch(err => {
+                            logger.error("processUpdateAsset Failed" + err);
+                            return reject(err)
+                        })
+                })
+                .catch(error => {
+                    logger.error("No records found for updating. Asset needs to be created first");
+                    return reject(error);
+                })
+        }
+    });
+}
+
+var processCreateAsset = function (eventPayload, configData, authToken) {
+    return new Promise((resolve, reject) => {
+        var svcContext = JSON.parse(eventPayload.SERVICE_CONTEXT.S);
+        logger.debug("svcContext: " + JSON.stringify(svcContext));
+
+        var assetApiPayload = {
+            "environment": svcContext.environment,
+            "service": eventPayload.SERVICE_NAME.S,
+            "created_by": svcContext.created_by,
+            "provider": svcContext.provider,
+            "provider_id": svcContext.provider_id,
+            "tags": svcContext.tags,
+            "domain": svcContext.domain,
+            "type": svcContext.type
+        };
+        if (_.includes(Object.keys(configData.EVENTS.EVENT_STATUS), eventPayload.EVENT_STATUS.S)) {
+            assetApiPayload["status"] = configData.EVENTS.EVENT_STATUS[eventPayload.EVENT_STATUS.S]
+        } else {
+            logger.error("Error in creating assets. Invalid status value in the payload");
+            return reject({
+                "error": "Error in creating assets. Invalid status value in the payload",
+                "details": eventPayload.EVENT_STATUS.S
+            });
+        }
+        logger.debug("assetApiPayload" + JSON.stringify(assetApiPayload));
+        var svcPayload = {
+            uri: configData.BASE_API_URL + configData.ASSETS_API_RESOURCE,
+            method: "POST",
+            headers: { Authorization: authToken },
+            json: assetApiPayload,
+            rejectUnauthorized: false
+        };
+
+        logger.debug("svcPayload" + JSON.stringify(svcPayload));
+        request(svcPayload, function (error, response, body) {
+            if (response.statusCode && response.statusCode === 200 && body && body.data) {
```

Comment:
are you sure body will always be json. what is body is string, then body.data will be undefined.
Same for other places in your code

Reply:
This is a request response and we also the know API we are hitting and its responses back.  HTTP response will have fixed formats for success and errors.